### DISCUSSION TITLE
How to use fn with arrow keys?

### DISCUSSION THREAD

**Discussion by Demianeen** [asmvik/yabai#2037] [Q&A] [answered]
I would like to set prev next space keys on the fn left and right arrows. I came up with something like this:
fn - left : yabai -m space --focus prev
fn - right : yabai -m space --focus next

It is not working when I press the fn + left arrow key. But works when I press ONLY the left arrow key. Is that intended behaviour?

**Comment by asmvik** [comment, marked as answer]
The `fn` modifier is implicitly set in the`arrow` keys  events (by macOS), so it is not possible to use it as a modifier with the arrow keys.

Edit:

```
The following keys can not be used with the fn-modifier:

"delete"     (kVK_ForwardDelete)
"home"       (kVK_Home)
"end"        (kVK_End)
"pageup"     (kVK_PageUp)
"pagedown"   (kVK_PageDown)
"insert"     (kVK_Help)
"left"       (kVK_LeftArrow)
"right"      (kVK_RightArrow)
"up"         (kVK_UpArrow)
"down"       (kVK_DownArrow)
"f1"         (kVK_F1)
"f2"         (kVK_F2)
"f3"         (kVK_F3)
"f4"         (kVK_F4)
"f5"         (kVK_F5)
"f6"         (kVK_F6)
"f7"         (kVK_F7)
"f8"         (kVK_F8)
"f9"         (kVK_F9)
"f10"        (kVK_F10)
"f11"        (kVK_F11)
"f12"        (kVK_F12)
"f13"        (kVK_F13)
"f14"        (kVK_F14)
"f15"        (kVK_F15)
"f16"        (kVK_F16)
"f17"        (kVK_F17)
"f18"        (kVK_F18)
"f19"        (kVK_F19)
"f20"        (kVK_F20)
 
"sound_up"          (NX_KEYTYPE_SOUND_UP)
"sound_down"        (NX_KEYTYPE_SOUND_DOWN)
"mute"              (NX_KEYTYPE_MUTE)
"play"              (NX_KEYTYPE_PLAY)            
"previous"          (NX_KEYTYPE_PREVIOUS)        
"next"              (NX_KEYTYPE_NEXT)
"rewind"            (NX_KEYTYPE_REWIND)          
"fast"              (NX_KEYTYPE_FAST)            
"brightness_up"     (NX_KEYTYPE_BRIGHTNESS_UP)
"brightness_down"   (NX_KEYTYPE_BRIGHTNESS_DOWN)
"illumination_up"   (NX_KEYTYPE_ILLUMINATION_UP) 
"illumination_down" (NX_KEYTYPE_ILLUMINATION_DOWN)
```


https://github.com/koekeishiya/skhd/issues/1

**Comment by Bellavene** [comment]
It is possible with bettertouchtools, but I highly suggest not to, because fn+arrow keys are the standard keys for HOME/END/PAGE UP/PAGE DOWN, or other navigation in apps.

**Comment by Bellavene** [comment]
Moreover, just as information, you actually can redefine keyboard keys by macos itself with hidutil. Very useful for custom keyboard layouts as described [here](https://www.naseer.dev/post/hidutil/). And [here](https://hidutil-generator.netlify.app) you can easily generate  a configuration plist file. Don't forget to add /private/etc/sudoers.d/hidutil (the same way you do it for yabai's scripting additions) on latest macos systems, otherwise it won't work.
